packages/drivers/odsp-driver/src/odspCache.ts

```diff
@@ -58,15 +59,74 @@ class GarbageCollector<TKey> {
     }
 }
 
+class MapWithExpiration<TKey, TValue> extends Map<TKey, TValue> implements IDisposable {
+    public disposed: boolean = false;
+    private readonly expirationTimeouts = new Map<TKey, ReturnType<typeof setTimeout>>();
+
```

Comment:
I like this approach too - but if we could simply add the current behavior to PromiseCache, I think it would probably make more sense.